### PR TITLE
Text editor: show current line number

### PR DESCRIPTION
**Pull request by hius07**
Dioscussed in https://github.com/koreader/koreader/issues/12618. Closes https://github.com/koreader/koreader/issues/12618.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/12632)

### CODE REVIEW

**Review by Frenzie — APPROVED** [review]
Fine by me

**Comment by poire-z** [comment]
The lines symbols feels a little wide.
Dunno if using some narrower vertical dots like in the first column of https://www.w3.org/TR/xml-entity-names/025.html (U+0256/7/A/B) would look ok.

**Comment by poire-z** [comment]
The dots are a bit heavy to see the space in between, but I still prefer that, it's less "in your face".
(I'd also may be prefer it in bold like the other buttons, it may feel less patchworky.)

**Comment by NiLuJe** [comment]
Both would work for me, FWIW.

**Comment by Commodore64user** [comment]
I think the original proposal is better but who am I to face @poire-z?

**Comment by hius07** [comment]
Four vertical dots. Non-bold, because bold dots become just vertical line in my eink. Also, the lines in the next four buttons (arrows) are thin, look similar to the non-bold digits in the line number.

**Comment by jonnyl2** [comment]
Tested, and I think it looks very nice. My immediate first impression was also that the line number looks a bit skinny, but I got used to it quick and I think that notionally it makes sense that this button's style is a bit different than the others, because it shows an info value and not a direct action to call.

**Comment by mergen3107** [comment]
@hius07 
Thank you, it works great!

Only thing I noticed is that when I go to other line, the line number is rendered with jagged edges. It renders normal when I refresh manually with a short diagonal swipe. See photos below.

I don't know if that's an issue or limitation in that context.

Thanks!

Before manual refresh:
image

After manual refresh
image

**Comment by hius07** [comment]
> the line number is rendered with jagged edges

@NiLuJe can you look at this please. On every key strike that changes the line number I use
https://github.com/koreader/koreader/blob/ce7fcff3608cd3f4ee24898b82a3c2e680f202c3/frontend/ui/widget/button.lua#L512-L526

Navigation buttons use
https://github.com/koreader/koreader/blob/ce7fcff3608cd3f4ee24898b82a3c2e680f202c3/frontend/ui/widget/inputdialog.lua#L722-L733

**Comment by NiLuJe** [comment]
Double-check the (verbose) logs to see what kind of refresh is actually used, but this appears to be requesting `"ui"`, which is perfectly fine, so it just smells like an overzealous optimization on the Scribe's part ;).

**Comment by NiLuJe** [comment]
By which I mean, it'll behave exactly like a VirtualKeyboard key, so, expect all kinds of funkiness, but the current approach is (potentially) the least likely to trip even worse behavior ;).

File: frontend/ui/widget/inputdialog.lua [hunk 1]
```diff
@@ -961,15 +966,17 @@ function InputDialog:_addScrollButtons(nav_bar)
...
             -- Add a button to go to the line by its number in the file
             table.insert(row, {
-                text = _("Go"),
+                text = "", -- current line number
+                font_bold = false,
+                id = "go",
                 callback = function()
                     self:toggleKeyboard(false) -- hide text editor keyboard
-                    local cur_line_num, last_line_num = self._input_widget:getLineNums()
+                    local curr_line_num, last_line_num = self._input_widget:getLineNums()
```
Comment (poire-z) [inline frontend/ui/widget/inputdialog.lua:974]: Not certain we have consistency and that one is preferable to the other:
```
$ rgrep ' cur_' frontend plugins  | wc -l
270

$ rgrep ' curr_' frontend plugins  | wc -l
124
```

**Review by poire-z — APPROVED** [review]
We'll see how bothering the button refresh issue is I guess.

File: frontend/ui/widget/inputdialog.lua (hunk 1 above)
Comment (Frenzie) [reply frontend/ui/widget/inputdialog.lua:974]: I'd personally stick with `cur_` but it doesn't seem very important (unless it's too close to each other).

**Pull request merged**